src/commands/mapsheet-coverage/mapsheet.coverage.ts

```diff
@@ -0,0 +1,224 @@
+import { gzipSync } from 'node:zlib';
+
+import { ConfigTileSetRaster } from '@basemaps/config';
+import { EpsgCode } from '@basemaps/geo';
+import { fsa } from '@chunkd/fs';
+import { command, number, option, optional, string } from 'cmd-ts';
+import { basename } from 'path/posix';
+import pc from 'polygon-clipping';
+import { StacCollection } from 'stac-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { MapSheet } from '../../utils/mapsheet.js';
+import { config, registerCli, verbose } from '../common.js';
+
+/** Datasets to skip */
+const Skip = new Set([
+  /** This covers the entire country and can be ignored */
+  'new-zealand_2012_dem_8m',
+]);
+
+/** Location for the output files to be stored */
+const OutputPath = '/tmp/mapsheet-coverage/';
+
+/**
+ * Number of decimal places to restrict capture areas too
+ * Rough numbers of decimal places to precision in meters
+ *
+ * 5DP - 1m
+ * 6DP - 0.1m
+ * 7DP - 0.01m (1cm)
+ * 8DP - 0.001m (1mm)
+ */
+const TruncationFactor = 8;
+
+/**
+ * Truncate a multi polygon in lat,lng to {@link TruncationFactor} decimal places
+ *
+ * @warning This destroys the source geometry
+ * @param polygons
+ */
+function truncateGeoJson(feature: GeoJSON.Feature): asserts feature is GeoJSON.Feature<GeoJSON.MultiPolygon> {
+  const factor = Math.pow(10, TruncationFactor);
+
+  const geom = feature.geometry;
+  // force polygons to be multipolygons
+  if (geom.type === 'Polygon') feature.geometry = { type: 'MultiPolygon', coordinates: [geom.coordinates] };
+
+  // Only multipolygons can be truncated
+  if (feature.geometry.type !== 'MultiPolygon') throw new Error('Unable to truncate: ' + feature.geometry?.type);
+
+  const multiPoly = feature.geometry;
+
+  for (const poly of multiPoly.coordinates) {
+    for (const ring of poly) {
+      for (const pt of ring) {
+        pt[0] = Math.round(pt[0]! * factor) / factor;
+        pt[1] = Math.round(pt[1]! * factor) / factor;
+      }
+    }
+  }
+}
+
+/** allow the configuration layer choice between 2193 and 3857 */
+const ValidCodes = new Set([EpsgCode.Google, EpsgCode.Nztm2000]);
+
+export const commandMapSheetCoverage = command({
+  name: 'mapsheet-coverage',
+  description: 'Create a list of mapsheets needing to be created from a basemaps configuration',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+    epsgCode: option({
+      type: number,
+      long: 'epsg-code',
+      description: 'Basemaps configuration layer ESPG code to use',
+      defaultValueIsSerializable: true,
+      defaultValue: () => EpsgCode.Nztm2000,
+    }),
+    location: option({
+      type: string,
+      long: 'location',
+      description: 'Location of the basemaps configuration file',
```

Comment:
Should this mention that it's a URL?

Reply:
this could be a URL or a local location eg "./elevation.json"